### PR TITLE
Fix ES->MA connection

### LINKED ISSUE #5645: Wrong data in Spain<->Morocco connection
## Description
Just as I suspected, the data for the Spain<->Morocco connection is wrong, as can be seen on the [ESIOS webpage](https://www.esios.ree.es/en/analysis/10209) (ignore the lack of data during last night, that's not linked to this).
It is related to the fact that the parser returns MWh of fifteen minutes intervals, while the backend expects MW, so averages instead of adding.
Apart from changing something in the backend, the easiest solution that I found is forcing the API to return hour intervals. We would have to add a small delay also, to prevent returning incomplete data. (e.g. server only adds 11:00 and 11:15 if the provided date is 11:24)
We could alternatively multiply the 15 minutes interval data by 4 to get the MW. This would increase the granularity but would imply adding specific cases for this exchange, the parser would break if the granularity ever changes, etc. Whereas by forcing the granularity to 1 hour we make sure the granularity is always correct.
What is your opinion?

### PR DESCRIPTION
**Pull request by silimotion**
## Issue

Closes #5645 

## Description

I added a new mapping to take into account the granularity of the data, as discussed in #5645

### Double check

- [x] I have tested my parser changes locally with `poetry run test_parser "zone_key"`
- [x] I have run `pnpx prettier --write .` and `poetry run format` to format my changes.

### CODE REVIEW

File: parsers/ESIOS.py [hunk 1]
```diff
@@ -55,7 +61,10 @@ def fetch_exchange(
     }
 
     zone_key = ZoneKey("->".join(sorted([zone_key1, zone_key2])))
-    if zone_key not in EXCHANGE_ID_MAP.keys():
+    if (
+        zone_key not in EXCHANGE_ID_MAP.keys()
+        or zone_key not in EXCHANGE_GRANULARITY_FACTOR_MAP.keys()
```
Comment (unitrium) [inline parsers/ESIOS.py:66]:
```suggestion
        or zone_key not in EXCHANGE_GRANULARITY_FACTOR_MAP
```
No need for the `keys` method, you can directly make sure it's here.

File: parsers/ESIOS.py [hunk 2]
```diff
@@ -55,7 +61,10 @@ def fetch_exchange(
     }
 
     zone_key = ZoneKey("->".join(sorted([zone_key1, zone_key2])))
-    if zone_key not in EXCHANGE_ID_MAP.keys():
+    if (
+        zone_key not in EXCHANGE_ID_MAP.keys()
```
Comment (unitrium) [inline parsers/ESIOS.py:65]:
```suggestion
        zone_key not in EXCHANGE_ID_MAP
```

File: parsers/ESIOS.py [hunk 3]
```diff
@@ -24,6 +24,12 @@
     "ES->MA": "10209",
 }
 
+# Map each exchange to the needed factor to adjust from MWh to MW. Depends on the time granularity of the API for each request
+EXCHANGE_GRANULARITY_FACTOR_MAP = {
+    "AD->ES": 1,
+    "ES->MA": 4,
```
Comment (unitrium) [inline parsers/ESIOS.py:30]: Could you just add a comment that it's a 15min interval, so that when we look back at the code we understand this factor.

**Review by unitrium — COMMENTED** [review]
Thanks, could you just add a bit more comments so that we can better understand the mapping in the future? I will let @VIKTORVAV99 do the final review

File: parsers/ESIOS.py [hunk 4]
```diff
@@ -24,6 +24,13 @@
     "ES->MA": "10209",
 }
 
+# Map each exchange to the needed factor to adjust from MWh to MW. Depends on the time granularity of the API for each request
+# E.g ES->MA is 4 because the API returns 15 minutes intervals data (15 min = 1/4 of an hour; P=E/t).
+EXCHANGE_GRANULARITY_FACTOR_MAP = {
```
Comment (VIKTORVAV99) [inline parsers/ESIOS.py:29]:
```suggestion
EXCHANGE_MULTIPLICATION_FACTOR_MAP = {
```

File: parsers/ESIOS.py [hunk 5]
```diff
@@ -55,7 +62,10 @@ def fetch_exchange(
     }
 
     zone_key = ZoneKey("->".join(sorted([zone_key1, zone_key2])))
-    if zone_key not in EXCHANGE_ID_MAP.keys():
+    if (
+        zone_key not in EXCHANGE_ID_MAP
+        or zone_key not in EXCHANGE_GRANULARITY_FACTOR_MAP
```
Comment (VIKTORVAV99) [inline parsers/ESIOS.py:67]:
```suggestion
        or zone_key not in EXCHANGE_MULTIPLICATION_FACTOR_MAP
```

File: parsers/ESIOS.py [hunk 6]
```diff
@@ -82,6 +92,8 @@ def fetch_exchange(
             -value["value"] if zone_key.partition("->")[0] == "ES" else value["value"]
         )
 
+        net_flow *= EXCHANGE_GRANULARITY_FACTOR_MAP[zone_key]
```
Comment (VIKTORVAV99) [inline parsers/ESIOS.py:95]:
```suggestion
        net_flow *= EXCHANGE_MULTIPLICATION_FACTOR_MAP[zone_key]
```

**Review by VIKTORVAV99 — COMMENTED** [review]
Overall this looks good, but I would rename the mapping to better represent the value it contains, i.e. the multiplication factor, not the granularity itself.

**Review by VIKTORVAV99 — APPROVED** [review]
LGTM! 🎉
Thanks for implementing the fix.

**Pull request merged**